extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -215,29 +239,26 @@ export class AmpAutocomplete extends AMP.BaseElement {
       );
     }
 
-    if (
-      this.templates_.hasTemplate(this.element, 'template, script[template]')
-    ) {
-      this.templateElement_ = this.templates_.findTemplate(
-        this.element,
-        'template, script[template]'
-      );
-      // Dummy render to verify existence of "data-value" attribute.
-      this.templates_
-        .renderTemplate(this.templateElement_, /** @type {!JsonObject} */ ({}))
-        .then(renderedEl => {
-          userAssert(
-            renderedEl.hasAttribute('data-value') ||
-              renderedEl.hasAttribute('data-disabled'),
-            `${TAG} requires a "data-value" or "data-disabled" attribute.`
-          );
-        });
-    }
-
-    this.filter_ = userAssert(
-      this.element.getAttribute('filter'),
-      `${TAG} requires "filter" attribute.`
```

Comment:
`filter` doesn't default to "NONE" if the attribute is not present?

Reply:
No. The `filter` attribute specifies a type of client-side filtering, so it is better if the publisher explicitly identifies when they don't want that. If we default to some value silently, it could be hard to debug an undesirable behavior here.

Reply:
Should the attribute be `mandatory: true` in validation then?

Reply:
Yes, that sounds like a reasonable change to me. Would you prefer that in this PR or in a follow up? 

The reason that it wasn't originally marked as `mandatory` in the validations was because it was assumed it could be enforced here for both valid and invalid AMP pages at runtime.

Reply:
Follow-up is fine.

Reply:
https://github.com/ampproject/amphtml/pull/25798